src/managers/FileManager.ts

```diff
@@ -38,6 +37,70 @@ export class FileManager {
         return this.cache[lowerFilePath] as CodeFile;
     }
 
+    private getFunctionInfo(lines: string[]) {
```

Comment:
This function walks the source file and extracts every function declaration, whether it be at the root level, in an object, or a variable assignment.